plugin/kubernetes/controller.go

```diff
@@ -241,8 +263,56 @@ func namespaceListFunc(ctx context.Context, c kubernetes.Interface, s labels.Sel
 		if s != nil {
 			opts.LabelSelector = s.String()
 		}
-		listV1, err := c.CoreV1().Namespaces().List(ctx, opts)
-		return listV1, err
+		return c.CoreV1().Namespaces().List(ctx, opts)
+	}
+}
+
+func serviceWatchFunc(ctx context.Context, c kubernetes.Interface, ns string, s labels.Selector) func(options meta.ListOptions) (watch.Interface, error) {
```

Comment:
are these functions moved? Or why do we need these now?

Reply:
They are not new, they are moved from `watch.go`, which was originally held an implementation of the since removed Watcher interface. At some point, these completely unrelated (except by name only) k8s api WatchFunc functions snuck in there.  When the CoreDNS watch feature was removed, the Watcher implementation was removed, and all that was left were the k8s api WatchFunc.